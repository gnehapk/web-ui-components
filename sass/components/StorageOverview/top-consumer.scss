.kubevirt-top-consumer__line-chart {
  opacity: 1;
}

.kubevirt-top-consumer__time-duration {
  float: right;
  padding-right: 1rem;
  font-size: 0.8em;
}
