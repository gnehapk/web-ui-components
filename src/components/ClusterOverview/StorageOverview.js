import React from 'react';

import { Grid, GridItem } from '@patternfly/react-core';

import { Dashboard, DashboardBody, DashboardHeader } from '../Dashboard';

import { StorageDetailsConnected } from './Details/storageDetails'; // TODO: re-export

const LeftCards = () => (
  <GridItem key="left" lg={3} md={12} sm={12}>
    <Grid>
      <GridItem lg={12} md={6} sm={12}>
        <StorageDetailsConnected />
      </GridItem>
    </Grid>
  </GridItem>
);

export const StorageOverview = () => (
  <Dashboard>
    <DashboardHeader>Storage Overview</DashboardHeader>
    <DashboardBody>
      <Grid>
        <LeftCards />
      </Grid>
    </DashboardBody>
  </Dashboard>
);

StorageOverview.propTypes = {};
