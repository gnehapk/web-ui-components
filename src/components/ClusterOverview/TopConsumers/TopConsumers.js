import React from 'react';
import { ProgressBar, OverlayTrigger, Tooltip, Col, Row } from 'patternfly-react';
import PropTypes from 'prop-types';

import { getFormElement } from '../../Form';
import { DROPDOWN } from '../../Form/constants';
import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
  DashboardCardTitleHelp,
} from '../../Dashboard/DashboardCard';
import { ClusterOverviewContextGenericConsumer } from '../ClusterOverviewContext';
import { InlineLoading } from '../../Loading';

const ConsumersBar = ({ now, description, label }) => (
  <Row className="kubevirt-consumers__bar">
    <Col lg={4} md={4} sm={4} xs={4}>
      <OverlayTrigger
        overlay={<Tooltip id={`tooltip-for-${description}`}>{description}</Tooltip>}
        placement="top"
        trigger={['hover', 'focus']}
        rootClose={false}
      >
        <div className="kubevirt-consumers__bar-description">{description}</div>
      </OverlayTrigger>
    </Col>
    <Col lg={5} md={5} sm={5} xs={5} className="kubevirt-consumers__bar-column-progress">
      <div className="progress kubevirt-consumers__bar-progress">
        <ProgressBar min={0} max={100} now={now} key={1} isChild />
        <ProgressBar min={0} max={100} now={100 - now} key={2} bsClass="progress-bar progress-bar-remaining" isChild />
      </div>
    </Col>
    <Col lg={3} md={3} sm={3} xs={3} className="kubevirt-consumers__bar-label">
      <OverlayTrigger
        overlay={<Tooltip id={`tooltip-for-${label}`}>{label}</Tooltip>}
        placement="top"
        trigger={['hover', 'focus']}
        rootClose={false}
      >
        <span>{label}</span>
      </OverlayTrigger>
    </Col>
  </Row>
);

ConsumersBar.propTypes = {
  now: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

const getConsumers = metric => {
  const max = Math.max(...metric.consumers.map(c => c.usage));
  return metric.consumers
    .map(c => ({
      now: Math.round((100 * c.usage) / max),
      description: c.name,
      label: c.label,
    }))
    .sort((a, b) => b.now - a.now);
};

class TopConsumersBody extends React.PureComponent {
  constructor(props) {
    super(props);
    const typeMetrics = props.metrics[0].metrics;
    this.state = {
      type: props.metrics[0].title,
      sortBy: typeMetrics[Object.keys(typeMetrics)[0]].title,
    };
  }

  onDropdownChange = (key, newValue) => this.setState({ [key]: newValue });

  render() {
    const { metrics, LoadingComponent } = this.props;
    const currentMetricType = metrics.find(m => m.title === this.state.type);
    const currentMetricKey = Object.keys(currentMetricType.metrics).find(
      key => currentMetricType.metrics[key].title === this.state.sortBy
    );
    const currentMetric = currentMetricType.metrics[currentMetricKey];
    const sortByDropdown = {
      id: 'sort-by-dropdown',
      type: DROPDOWN,
      choices: Object.keys(currentMetricType.metrics).map(key => currentMetricType.metrics[key].title),
      onChange: newValue => this.onDropdownChange('sortBy', newValue),
      value: this.state.sortBy,
    };
    const typeDropdown = {
      id: 'type-dropdown',
      type: DROPDOWN,
      choices: metrics.map(m => m.title),
      onChange: newValue => this.onDropdownChange('type', newValue),
      value: this.state.type,
    };
    return (
      <div>
        <div className="kubevirt-consumers__filters">
          {getFormElement(typeDropdown)}
          {getFormElement(sortByDropdown)}
        </div>
        <div className="kubevirt-consumers__results">
          <Row className="kubevirt-consumers__description">
            <Col lg={4} md={4} sm={4} xs={4} className="kubevirt-consumers__metric-type">
              {this.state.type}
            </Col>
            <Col lg={8} md={8} sm={8} xs={8} className="kubevirt-consumers__metric-description">
              {currentMetric.description}
            </Col>
          </Row>
          {currentMetric && currentMetric.isLoading ? (
            <LoadingComponent />
          ) : (
            getConsumers(currentMetric).map((c, index) => (
              <ConsumersBar key={index} {...c} className="kubevirt-consumers__bar" />
            ))
          )}
        </div>
      </div>
    );
  }
}

TopConsumersBody.propTypes = {
  metrics: PropTypes.array.isRequired,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};

export const TopConsumers = ({ metrics, LoadingComponent }) => (
  <DashboardCard>
    <DashboardCardHeader>
      <DashboardCardTitle>Top Consumers</DashboardCardTitle>
      <DashboardCardTitleHelp>help for top consumers</DashboardCardTitleHelp>
    </DashboardCardHeader>
    <DashboardCardBody>
      <TopConsumersBody metrics={metrics} LoadingComponent={LoadingComponent} />
    </DashboardCardBody>
  </DashboardCard>
);

TopConsumers.defaultProps = {
  LoadingComponent: InlineLoading,
};

TopConsumers.propTypes = {
  metrics: PropTypes.array.isRequired,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

const TopConsumersConnected = () => (
  <ClusterOverviewContextGenericConsumer Component={TopConsumers} dataPath="consumersData" />
);

export default TopConsumersConnected;
