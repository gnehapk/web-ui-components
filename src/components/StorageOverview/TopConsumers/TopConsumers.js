import React from 'react';
import PropTypes from 'prop-types';

import { LineChart, Row, Col } from 'patternfly-react';
import { ChartGroup, ChartLine, ChartTheme, ChartLabel, ChartTooltip, Chart, ChartAxis, ChartLegend, ChartVoronoiContainer } from '@patternfly/react-charts';

import { InlineLoading } from '../../Loading';
import { formatToShortTime } from '../../../utils';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { StorageOverviewContext } from '../StorageOverviewContext/StorageOverviewContext';
import { getTopConsumerVectorStats } from '../../../selectors/prometheus/storage';

const TopConsumersBody = ({ topConsumerStats }) => {
  let results = 'No data available';

  if (topConsumerStats.length) {
    const stats = getTopConsumerVectorStats(topConsumerStats);
    const { chartData, legends, xAxisData, maxCapacity, unit } = stats;
    const yTickValues = [0, Number((maxCapacity / 4).toFixed(1)), Number((maxCapacity / 2).toFixed(1)), Number(((3 * maxCapacity) / 4).toFixed(1)), maxCapacity];

    const chartLineList = chartData.map((data, i) =>
      <ChartLine
        key={i}
        data={data}
        x={0}
        y={1}
      />
    );
    results = (
      <React.Fragment>
        <Row>
          <Col className="kubevirt-top-consumer__time-duration">Last 6 hours</Col>
        </Row>
        <Row>
          <div>
            <Chart
              domain={{ y: [0, maxCapacity] }}
              theme={ChartTheme.light.multi}
              height={175}
              padding={{ top: 30, bottom: 20, left: 40, right: 15 }}
              containerComponent={
                <ChartVoronoiContainer
                labels={datum => `${datum.y} ${unit}`}
                labelComponent={<ChartTooltip style={{ fontSize: 12, padding: 5 }} />}
                />
              }
            >
              <ChartGroup>
                {chartLineList}
              </ChartGroup>
              <ChartAxis tickValues={xAxisData} style={{ tickLabels: { fontSize: 8, padding: 5 } }} />
              <ChartAxis label={`Used Capacity(${unit})`} dependentAxis tickValues={yTickValues} style={{ tickLabels: { fontSize: 8, padding: 5 }, axisLabel: { fontSize: 8, padding: 25 }, }} />
            </Chart>
          </div>
          <div className="chart-legend">
            <ChartLegend
              data={legends}
              orientation="horizontal"
              gutter={10}
              height={30}
              padding={{ top: 5, bottom: 5, left: 20 }}
              theme={ChartTheme.light.multi}
              x={50}
              y={8}
            />
          </div>
        </Row>

      </React.Fragment>
    );
  }

  return <div>{results}</div>;
};

export const TopConsumers = ({ topConsumerStats, topConsumerLoaded, LoadingComponent }) => (
  <DashboardCard>
    <DashboardCardHeader>
      <DashboardCardTitle>Top Projects by Used Capacity</DashboardCardTitle>
    </DashboardCardHeader>
    <DashboardCardBody isLoading={!topConsumerLoaded} LoadingComponent={LoadingComponent}>
      <TopConsumersBody topConsumerStats={topConsumerStats} />
    </DashboardCardBody>
  </DashboardCard>
);

TopConsumersBody.propTypes = {
  topConsumerStats: PropTypes.array.isRequired,
};

TopConsumers.defaultProps = {
  topConsumerStats: [],
  topConsumerLoaded: false,
  LoadingComponent: InlineLoading,
};

TopConsumers.propTypes = {
  topConsumerStats: PropTypes.array,
  topConsumerLoaded: PropTypes.bool,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

const TopConsumersConnected = () => (
  <StorageOverviewContext.Consumer>{props => <TopConsumers {...props} />}</StorageOverviewContext.Consumer>
);

export default TopConsumersConnected;
