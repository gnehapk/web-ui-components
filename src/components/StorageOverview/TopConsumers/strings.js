export const PROJECTS = 'Projects';
export const STORAGE_CLASSES = 'Storage Classes';
export const PODS = 'Pods';

export const BY_USED_CAPACITY = 'By Used Capacity';
export const BY_REQUESTED_CAPACITY = 'By Requested Capacity';
