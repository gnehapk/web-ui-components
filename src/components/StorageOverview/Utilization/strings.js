export const ONE_HR = '1 Hour';
export const SIX_HR = '6 Hours';
export const TWENTY_FOUR_HR = '24 Hours';
