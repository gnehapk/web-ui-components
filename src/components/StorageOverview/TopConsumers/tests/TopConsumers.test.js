import React from 'react';
import { shallow, mount } from 'enzyme';

import TopConsumersConnected, { TopConsumers, TopConsumersBody } from '../TopConsumers';
import { default as TopConsumersFixtures } from '../fixtures/TopConsumers.fixture';
import { default as StorageOverviewFixtures } from '../../fixtures/StorageOverview.fixture';
import { StorageOverviewContext } from '../../StorageOverviewContext';
import { selectDropdownItem } from '../../../../tests/enzyme';

import { PROJECTS, STORAGE_CLASSES, PODS, BY_USED_CAPACITY, BY_REQUESTED_CAPACITY } from '../strings';

// eslint-disable-next-line react/prop-types
const testTopConsumersOverview = ({ props }) => <TopConsumers {...props} />;

const getTypeDropdown = component => component.find('#metric-type');
const getSortByDropdown = component => component.find('#sort-by');

const testConsumerResults = (component, type, sortBy) => {
  selectDropdownItem(getTypeDropdown(component), type);
  selectDropdownItem(getSortByDropdown(component), sortBy);
  const consumerBody = component.find(TopConsumersBody);

  expect(consumerBody.props().metricType).toBe(type);
  expect(consumerBody.props().sortByOption.name).toBe(sortBy);
};

describe('<TopConsumers />', () => {
  TopConsumersFixtures.forEach(fixture => {
    it(`renders ${fixture.name} correctly`, () => {
      const component = shallow(testTopConsumersOverview(fixture));
      expect(component).toMatchSnapshot();
    });
  });
  it('switches between metrics', () => {
    const component = mount(testTopConsumersOverview(TopConsumersFixtures[0]));
    // default metric is workload sorted by Projects
    const consumerBody = component.find(TopConsumersBody);
    expect(consumerBody.props().metricType).toBe(PROJECTS);

    testConsumerResults(component, PROJECTS, BY_USED_CAPACITY);
    testConsumerResults(component, PROJECTS, BY_REQUESTED_CAPACITY);
    testConsumerResults(component, STORAGE_CLASSES, BY_USED_CAPACITY);
    testConsumerResults(component, STORAGE_CLASSES, BY_REQUESTED_CAPACITY);
    testConsumerResults(component, PODS, BY_USED_CAPACITY);
    testConsumerResults(component, PODS, BY_REQUESTED_CAPACITY);
  });
  it('renders correctly with Provider', () => {
    const component = shallow(
      <StorageOverviewContext.Provider value={StorageOverviewFixtures[0].props}>
        <TopConsumersConnected />
      </StorageOverviewContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
