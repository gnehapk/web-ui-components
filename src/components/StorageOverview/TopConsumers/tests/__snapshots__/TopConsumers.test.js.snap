// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopConsumers /> data-renders-correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title kubevirt-dashboard__card-title"
    >
      Top Projects by Used Capacity
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <div>
      <div
        class="row"
      >
        <div
          class="kubevirt-top-consumer__time-duration"
        >
          Last 6 hours
        </div>
      </div>
      <div
        class="row"
      >
        <div>
          <div
            class="VictoryContainer"
            style="height:100%;width:100%;user-select:none;pointer-events:none;touch-action:none;position:relative"
          >
            <svg
              aria-labelledby="victory-container-1-title victory-container-1-desc"
              height="175"
              role="img"
              style="pointer-events:all;width:100%;height:100%"
              viewBox="0 0 451 175"
              width="451"
            >
              <g>
                <g
                  clip-path="url(#victory-clip-2)"
                  style="height:100%;width:100%;user-select:none"
                >
                  <defs>
                    <clipPath
                      id="victory-clip-2"
                    >
                      <rect
                        height="125"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                        width="396"
                        x="40"
                        y="30"
                      />
                    </clipPath>
                  </defs>
                  <path
                    d="M40,30L119.2,30L198.4,30L277.6,30L356.8,30L436,30"
                    role="presentation"
                    shape-rendering="auto"
                    style="fill:none;stroke:#39a5dc;width:25px;stroke-width:2"
                  />
                </g>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke:#72767b;fill:transparent;stroke-width:1;stroke-linecap:round;stroke-linejoin:round"
                  vector-effect="non-scaling-stroke"
                  x1="40"
                  x2="436"
                  y1="155"
                  y2="155"
                />
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="40"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="40"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="40"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="40"
                    >
                      17:05
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="119.2"
                    x2="119.2"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="119.2"
                    x2="119.2"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="119.2"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="119.2"
                    >
                      17:10
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="198.4"
                    x2="198.4"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="198.4"
                    x2="198.4"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="198.4"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="198.4"
                    >
                      17:15
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="277.6"
                    x2="277.6"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="277.6"
                    x2="277.6"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="277.6"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="277.6"
                    >
                      17:20
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="356.8"
                    x2="356.8"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="356.8"
                    x2="356.8"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="356.8"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="356.8"
                    >
                      17:25
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="436"
                    x2="436"
                    y1="155"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="436"
                    x2="436"
                    y1="155"
                    y2="156"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="6.84"
                    x="436"
                    y="161"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="middle"
                      x="436"
                    >
                      17:30
                    </tspan>
                  </text>
                </g>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke:#72767b;fill:transparent;stroke-width:1;stroke-linecap:round;stroke-linejoin:round"
                  vector-effect="non-scaling-stroke"
                  x1="40"
                  x2="40"
                  y1="30"
                  y2="155"
                />
                <text
                  direction="inherit"
                  dx="0"
                  dy="-1.1600000000000001"
                  transform="rotate(-90,15,92.5)"
                  x="15"
                  y="92.5"
                >
                  <tspan
                    dx="0"
                    style="font-size:8px;padding:25px;text-anchor:middle;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                    text-anchor="middle"
                    x="15"
                  >
                    Used Capacity(Mi)
                  </tspan>
                </text>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="436"
                    y1="155"
                    y2="155"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="39"
                    y1="155"
                    y2="155"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="2.84"
                    x="34"
                    y="155"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="end"
                      x="34"
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="436"
                    y1="123.76419809177646"
                    y2="123.76419809177646"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="39"
                    y1="123.76419809177646"
                    y2="123.76419809177646"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="2.84"
                    x="34"
                    y="123.76419809177646"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="end"
                      x="34"
                    >
                      11
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="436"
                    y1="92.52839618355293"
                    y2="92.52839618355293"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="39"
                    y1="92.52839618355293"
                    y2="92.52839618355293"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="2.84"
                    x="34"
                    y="92.52839618355293"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="end"
                      x="34"
                    >
                      22
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="436"
                    y1="61.29259427532941"
                    y2="61.29259427532941"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="39"
                    y1="61.29259427532941"
                    y2="61.29259427532941"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="2.84"
                    x="34"
                    y="61.29259427532941"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="end"
                      x="34"
                    >
                      33
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:none;fill:none;pointer-events:painted"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="436"
                    y1="30"
                    y2="30"
                  />
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke:transparent;fill:transparent;size:1px"
                    vector-effect="non-scaling-stroke"
                    x1="40"
                    x2="39"
                    y1="30"
                    y2="30"
                  />
                  <text
                    direction="inherit"
                    dx="0"
                    dy="2.84"
                    x="34"
                    y="30"
                  >
                    <tspan
                      dx="0"
                      style="font-size:8px;padding:5px;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;letter-spacing:normal;fill:#282d33;stroke:transparent"
                      text-anchor="end"
                      x="34"
                    >
                      44.02
                    </tspan>
                  </text>
                </g>
              </g>
            </svg>
            <div
              style="z-index:99;position:absolute;top:0;left:0;width:100%;height:100%"
            >
              <svg
                height="175"
                style="overflow:visible;width:100%;height:100%"
                viewBox="0 0 451 175"
                width="451"
              />
            </div>
          </div>
        </div>
        <div
          class="chart-legend"
        >
          <div
            class="VictoryContainer"
            style="pointer-events:none;touch-action:none;position:relative;width:450px;height:30px"
          >
            <svg
              aria-labelledby="victory-container-3-title victory-container-3-desc"
              height="30"
              role="img"
              style="pointer-events:all;width:450px;height:30px"
              width="450"
            >
              <rect
                height="28.700000000000003"
                role="presentation"
                shape-rendering="auto"
                style="fill:none"
                vector-effect="non-scaling-stroke"
                width="164.21538461538458"
                x="50"
                y="8"
              />
              <path
                d="M 59.128, 26.872
      h9.744
      v-9.744
      h-9.744
      z"
                role="presentation"
                shape-rendering="auto"
                style="type:square;fill:#39a5dc"
              />
              <text
                direction="inherit"
                dx="0"
                dy="4.97"
                x="80.8"
                y="22"
              >
                <tspan
                  dx="0"
                  style="font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;font-size:14px;letter-spacing:normal;padding:10px;fill:#282d33;stroke:transparent"
                  text-anchor="start"
                  x="80.8"
                >
                  openshift-namespace
                </tspan>
              </text>
            </svg>
            <div
              style="z-index:99;position:absolute;top:0;left:0;width:450px;height:30px"
            >
              <svg
                height="30"
                style="overflow:visible;width:450px;height:30px"
                width="450"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TopConsumers /> no-data-renders-correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title kubevirt-dashboard__card-title"
    >
      Top Projects by Used Capacity
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <div>
      No data available
    </div>
  </div>
</div>
`;

exports[`<TopConsumers /> render-correctly-on-loading 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title kubevirt-dashboard__card-title"
    >
      Top Projects by Used Capacity
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <div>
      <div
        class="spinner spinner-md blank-slate-pf-icon"
      />
    </div>
  </div>
</div>
`;
